flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -700,11 +752,20 @@ private <T> WatermarkGaugeExposingOutput<StreamRecord<T>> createOutputCollector(
             // If the chaining output does not copy we need to copy in the broadcast output,
             // otherwise multi-chaining would not work correctly.
             if (containingTask.getExecutionConfig().isObjectReuseEnabled()) {
-                return closer.register(new CopyingBroadcastingOutputCollector<>(asArray));
+                result = closer.register(new CopyingBroadcastingOutputCollector<>(asArray));
             } else {
-                return closer.register(new BroadcastingOutputCollector<>(asArray));
+                result = closer.register(new BroadcastingOutputCollector<>(asArray));
+            }
+        }
+
+        if (shouldAddMetric) {
```

Comment:
I'd suggest to add an inline comment explaining that this only happens when there's no downstream chained operators recording the metrics.

Reply:
Thanks for the suggestion. I have updated the PR to explain it.